### PR TITLE
Use /charm/ root, update pebble, stream bin copies.

### PR DESCRIPTION
**Pull request by hpidcock**
- Use /charm/ as the root for charm bins and sockets.
- Use latest "juju-fake-init" as the pebble.
- Stream pebble, jujuc and k8sagent binary copies.

## QA steps

Unit tests.
Deploy new k8s charm (examples here https://github.com/hpidcock/testing-charms)

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: cmd/k8sagent/unit/agent.go [hunk 1]
```diff
@@ -60,8 +61,10 @@ type k8sUnitAgent struct {
 	prometheusRegistry *prometheus.Registry
 
 	fileReaderWriter utils.FileReaderWriter
+	environment      utils.Environment
 
 	charmModifiedVersion int
+	envVars              envVars
```
Comment (wallyworld) [inline cmd/k8sagent/unit/agent.go:67]: We have StringMap in juju/cmd already which may suit

File: cmd/k8sagent/utils/environment.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// NewEnvironment returns a new Environment.
+func NewEnvironment() Environment {
+	return &environment{}
+}
+
+type environment struct{}
+
+var _ Environment = (*environment)(nil)
+
+func (environment) Setenv(k, v string) error {
```
Comment (wallyworld) [inline cmd/k8sagent/utils/environment.go:26]: should comment these

**Review by wallyworld — APPROVED** [review]
Great thanks

**Comment by hpidcock** [comment]
$$merge$$

**Pull request merged**